lms/djangoapps/discussion_api/api.py

```diff
@@ -72,6 +74,19 @@ def _get_thread_and_context(request, thread_id, parent_id=None, retrieve_kwargs=
         raise Http404
 
 
+def _user_is_author_or_privileged(request, cc_thread, context):
```

Comment:
I think it would be nicer to make this _is_user_author_or_privileged and return a boolean so it can be used by _get_thread_editable_fields